ui/page/components/wallet_dropdown.go

```diff
@@ -0,0 +1,264 @@
+	accountsResult, err := wal.GetAccountsRaw()
+	if err != nil {
+		log.Errorf("Error getting accounts: %s", err)
+		return 0, 0
+	}
+	var tBal, sBal int64
+	for _, account := range accountsResult.Accounts {
+		// If the wallet is watching-only, the spendable balance is zero.
+		if wal.IsWatchingOnlyWallet() {
+			account.Balance.Spendable = wal.ToAmount(0)
+		}
+		tBal += account.Balance.Total.ToInt()
+		sBal += account.Balance.Spendable.ToInt()
+	}
+	return tBal, sBal
+}
+
+func (d *WalletDropdown) getWalletItemLayout(wallet sharedW.Asset) layout.Widget {
+	return func(gtx C) D {
+		totalBal, spendable := d.walletBalance(wallet)
+		return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+			layout.Rigid(func(gtx C) D {
+				return layout.Flex{Axis: layout.Horizontal, Spacing: layout.SpaceBetween}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						lbl := d.Theme.SemiBoldLabel(wallet.GetWalletName())
+						lbl.MaxLines = 1
+						lbl.TextSize = values.TextSizeTransform(d.IsMobileView(), values.TextSize16)
+						return lbl.Layout(gtx)
+					}),
+					layout.Rigid(func(gtx C) D {
+						return d.Theme.Label(values.TextSizeTransform(d.IsMobileView(), values.TextSize16), wallet.ToAmount(totalBal).String()).Layout(gtx)
+					}),
+				)
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layout.Flex{Axis: layout.Horizontal, Spacing: layout.SpaceBetween}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						spendableText := d.Theme.Label(values.TextSize14, values.String(values.StrLabelSpendable))
+						spendableText.Color = d.Theme.Color.GrayText2
+						return spendableText.Layout(gtx)
+					}),
+					layout.Rigid(func(gtx C) D {
+						return d.Theme.Label(values.TextSizeTransform(d.IsMobileView(), values.TextSize14), wallet.ToAmount(spendable).String()).Layout(gtx)
+					}),
+				)
+			}),
+		)
+	}
+}
+
+func (d *WalletDropdown) WalletValidator(walletIsValid func(sharedW.Asset) bool) *WalletDropdown {
+	d.walletIsValid = walletIsValid
+	return d
+}
+
+func (d *WalletDropdown) getWalletByID(walletID int) sharedW.Asset {
+	for _, wallet := range d.allWallets {
+		if wallet.GetWalletID() == walletID {
+			return wallet
+		}
+	}
+	return nil
+}
+
+func (d *WalletDropdown) onChanged() {
+	if d.dropdown != nil {
+		walletID, err := strconv.Atoi(d.dropdown.Selected())
+		if err == nil {
+			wallet := d.getWalletByID(walletID)
+			if wallet != nil {
+				d.selectedWallet = wallet
+				if d.walletChangedCallback != nil {
+					d.walletChangedCallback(wallet)
+				}
+			}
+		}
+	}
+}
+
+func (d *WalletDropdown) SelectedWallet() sharedW.Asset {
+	return d.selectedWallet
+}
+
+func (d *WalletDropdown) SetChangedCallback(callback func(sharedW.Asset)) *WalletDropdown {
+	d.walletChangedCallback = callback
+	return d
+}
+
+func (d *WalletDropdown) Handle(gtx C) {
+	if d.dropdown.Changed(gtx) {
+		d.onChanged()
+	}
+}
+
+func (d *WalletDropdown) Build() *WalletDropdown {
+	d.SetupWallet()
+	return d
+}
```

Comment:
why you don't return `*WalletDropdown` from `SetupWallet()` and remove this function ?